### PR TITLE
Image editor fix for fabric.js API changes

### LINKED ISSUE #1737: imageEdit.js and fabric.js API fix
With updating the version of fabric.js, the API has changed and imageEdit.js needs to be fixed.

### PR DESCRIPTION
**Pull request by jonathanaustin**
Fixed imageEdit.js to handle new fabric.js API and provided new image editor examples. Closes #1737.

### CODE REVIEW

File: wcomponents-examples/src/main/resources/com/github/bordertech/wcomponents/examples/initImageEditExample.txt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<script type="text/javascript">
+
+// Custom editor config turning on redact
+require(["wc/config"], function(config) {
+  config.set({redact:true, crop:false, format:'jpg'}, "wc/ui/imageEdit");
+});
+
+// When using a static WImage, you need to link the imageEditor and fileUploader
```
Comment (ricksbrown) [inline wcomponents-examples/src/main/resources/com/github/bordertech/wcomponents/examples/initImageEditExample.txt:8]: As discussed this looks like something that needs to be tidied up, it should happen automagically.

File: wcomponents-theme/src/main/js/wc/ui/imageEdit.js [hunk 2]
```diff
@@ -449,8 +449,10 @@ function(has, mixin, wcconfig, Widget, event, classList, timers, prompt, i18n, f
 			});
 
 			fbCanvas._activeObject = null;
-
-			fbCanvas.setActiveGroup(group.setCoords()).renderAll();
+			// remove all objects and re-render
+			fbCanvas.clear().renderAll();
```
Comment (ricksbrown) [inline wcomponents-theme/src/main/js/wc/ui/imageEdit.js:453]: Wow that's what fabric makes you do to select all objects now? You have to remove and add? Are you sure?

This looks pretty recent http://jsfiddle.net/softvar/mRA8Z/

Still, if that's what you gotta do then that's what you gotta do. I wouldn't know, it's a FabricJS thing.

File: wcomponents-theme/src/main/js/wc/ui/imageEdit.js [hunk 3]
```diff
@@ -1275,8 +1280,8 @@ function(has, mixin, wcconfig, Widget, event, classList, timers, prompt, i18n, f
 				} else {
 					object = unscale(fbImage);
 					toDataUrlParams = {
-						left: object.getLeft(),
-						top: object.getTop(),
+						left: object.get('left'),
```
Comment (ricksbrown) [inline wcomponents-theme/src/main/js/wc/ui/imageEdit.js:1283]: Yeah this seems to be the way the FabricJS API is going

**Comment by ricksbrown** [comment]
Sorry it took me so long to review this. It looks really good, the new examples are gold.

Ultimately if it works it works. FabricJS seems to change quite a lot over time so it's hard to be confident in the API.

File: wcomponents-theme/src/main/js/wc/ui/imageEdit.js (hunk 2 above)
Comment (jonathanaustin) [reply wcomponents-theme/src/main/js/wc/ui/imageEdit.js:453]: You inspired me to take another look at this... and I found this doco from fabric about the breaking API http://fabricjs.com/v2-breaking-changes-2. I will make a change to use the ActiveSelection.

File: wcomponents-examples/src/main/resources/com/github/bordertech/wcomponents/examples/initImageEditExample.txt (hunk 1 above)
Comment (jonathanaustin) [reply wcomponents-examples/src/main/resources/com/github/bordertech/wcomponents/examples/initImageEditExample.txt:8]: Agree. Have raised issue #1743

Comment (jonathanaustin) [reply wcomponents-examples/src/main/resources/com/github/bordertech/wcomponents/examples/initImageEditExample.txt:8]: Agree. Have raised issue #1743

File: wcomponents-theme/src/main/js/wc/ui/imageEdit.js [hunk 4]
```diff
@@ -1180,9 +1186,19 @@ function(has, mixin, wcconfig, Widget, event, classList, timers, prompt, i18n, f
...
-			fbCanvas.setActiveGroup(group.setCoords()).renderAll();
+
+			// Create active selection from the objects in the group
+			var selection = new fabric.ActiveSelection(objects, {
+				canvas: fbCanvas
+			});
+
+			fbCanvas._activeObject = null;
+			fbCanvas.setActiveObject(selection);
+			fbCanvas.renderAll();
```
Comment (ricksbrown) [inline wcomponents-theme/src/main/js/wc/ui/imageEdit.js:1197]: Ah yes, this looks a lot more plausible. Good stuff.

File: wcomponents-theme/src/main/js/wc/ui/imageEdit.js [hunk 5]
```diff
@@ -447,10 +447,16 @@ function(has, mixin, wcconfig, Widget, event, classList, timers, prompt, i18n, f
...
 			});
+			
+			// Create active selection from the objects in the group
+			var selection = new fabric.ActiveSelection(objects, {
+				canvas: fbCanvas
+			});
 
 			fbCanvas._activeObject = null;
+			fbCanvas.setActiveObject(selection);
+			fbCanvas.renderAll();
```
Comment (ricksbrown) [inline wcomponents-theme/src/main/js/wc/ui/imageEdit.js:458]: Yep, looks bettererer.

**Comment by ricksbrown** [comment]
Nice, LGTM, good stuff.

**Pull request merged**